src/lib.rs

```diff
@@ -218,18 +218,29 @@ fn json_mget(ctx: &Context, args: Vec<String>) -> RedisResult {
         let results: Result<Vec<RedisValue>, RedisError> = keys
             .iter()
             .map(|key| {
-                let result = ctx
-                    .open_key(key)
+                ctx.open_key(key)
                     .get_value::<RedisJSON>(&REDIS_JSON_TYPE)?
                     .map(|doc| doc.to_string(&path, Format::JSON))
-                    .transpose()?;
-
-                Ok(result.into())
+                    .transpose()
+                    .map_or_else(|_| Ok(RedisValue::Null), |v| Ok(v.into()))
```

Comment:
It seems that current tests need to be updated accordingly.
Currently a failed key (`foo`) is tested in `testMgetCommand` in RedisJSON/tests/pytest/test.py

Now a failed path does not raise an exception, so this needs to be modified?
`r.expect('JSON.MGET', 'doc:0', 'doc:1', '42isnotapath').raiseError()`

And should probably add another test for combined successful and failed paths, such as:
```
    # Test that MGET fails on path errors
    r.cmd('DEL', 'test')
    r.assertOk(r.execute_command('JSON.SET', 'test', '.', '{"bull":4.2}'))
    raw = r.execute_command('JSON.MGET', 'doc:0', 'test', 'doc:1', '.bool')
    r.assertEqual(len(raw), 3)
    r.assertTrue(json.loads(raw[0]))
    r.assertEqual(raw[1], None)
    r.assertTrue(json.loads(raw[2]))

```